src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -796,47 +736,220 @@ private double getSubjectiveValueWeight(Criterion awt) {
-
 		switch (awt) {
 		case TELE:
-			avf.setTeleSubjectiveValueWeight(value);
+			this.setTeleSubjectiveValueWeight(value);
 			break;
 		case TERRACE:
-			avf.setTerraceSubjectiveValueWeight(value);
+			this.setTerraceSubjectiveValueWeight(value);
 			break;
 		case WIFI:
-			avf.setWifiSubjectiveValueWeight(value);
+			this.setWifiSubjectiveValueWeight(value);
 			break;
 		case FLOOR_AREA:
-			avf.setFloorAreaSubjectiveValueWeight(value);
+			this.setFloorAreaSubjectiveValueWeight(value);
 			break;
 		case FLOOR_AREA_TERRACE:
-			avf.setFloorAreaTerraceSubjectiveValueWeight(value);
+			this.setFloorAreaTerraceSubjectiveValueWeight(value);
 			break;
 		case NB_BATHROOMS:
-			avf.setNbBathroomsSubjectiveValueWeight(value);
+			this.setNbBathroomsSubjectiveValueWeight(value);
 			break;
 		case NB_BEDROOMS:
-			avf.setNbBedroomsSubjectiveValueWeight(value);
+			this.setNbBedroomsSubjectiveValueWeight(value);
 			break;
 		case NB_SLEEPING:
-			avf.setNbSleepingSubjectiveValueWeight(value);
+			this.setNbSleepingSubjectiveValueWeight(value);
 			break;
 		case NB_MIN_NIGHT:
-			avf.setNbMinNightSubjectiveValueWeight(value);
+			this.setNbMinNightSubjectiveValueWeight(value);
 			break;
 		case PRICE_PER_NIGHT:
-			avf.setPricePerNightSubjectiveValueWeight(value);
+			this.setPricePerNightSubjectiveValueWeight(value);
 			break;
 		default:
 			throw new IllegalArgumentException();
 		}
 
-		return avf;
+		return this;
 
 	}
 
+	/**
+	 * Gets the weight of the floor area subjective value
+	 * @return the weight of the attribute floorAreaSubjectiveValueWeight
+	 */
+	public double getFloorAreaSubjectiveValueWeight() {
+		return this.floorAreaSubjectiveValueWeight;
+	}
+
+	/**
+	 * Gets the interval of the floor area value. This interval represents the range of floor area the
+	 * user might accept
+	 * @return the attribute floorAreaValueFunction
+	 */
+	public PartialValueFunction<Double> getFloorAreaValueFunction() {
+		return this.floorAreaValueFunction;
+	}
+
+	/**
+	 * Gets the weight of the floor area Terrace subjective value
+	 * @return the weight of the attribute floorAreaTerraceSubjectiveValueWeight
+	 */
+	public double getFloorAreaTerraceSubjectiveValueWeight() {
+		return this.floorAreaTerraceSubjectiveValueWeight;
+	}
+
+	/**
+	 * Gets the interval of the floor area terrace value. This interval represents the range of floor area 
+	 * the user might accept
+	 * @return the attribute floorAreaTerraceValueFunction
+	 */
+	public PartialValueFunction<Double> getFloorAreaTerraceValueFunction() {
+		return this.floorAreaTerraceValueFunction;
+	}
+
+	/**
+	 * Gets the weight of the number of bathrooms subjective value
+	 * @return the weight of the attribute nbBathroomsSubjectiveValueWeight
+	 */
+	public double getNbBathroomsSubjectiveValueWeight() {
+		return this.nbBathroomsSubjectiveValueWeight;
+	}
+
+	/**
+	 * Gets an interval of the number of bathrooms. This interval represents the number of bathrooms the
```

Comment:
Ceci n’est pas correct : une PVF ne représente pas un intervalle de valeurs acceptables, au contraire, on essaye d’éviter cette approche dans ce projet.

Reply:
Comments modified in commit 19834bd1fc262b85dfde3bb64f31774ff46c8b1b